promslog/slog.go

```diff
@@ -38,115 +39,29 @@ const (
-			}
-		case slog.SourceKey, "caller":
-			if src, ok := a.Value.Any().(*slog.Source); ok {
-				a.Key = "caller"
-				switch callerAddFunc {
-				case true:
-					a.Value = slog.StringValue(filepath.Base(src.File) + "(" + filepath.Base(src.Function) + "):" + strconv.Itoa(src.Line))
-				default:
-					a.Value = slog.StringValue(filepath.Base(src.File) + ":" + strconv.Itoa(src.Line))
-				}
-			} else {
-				// If we can't cast the any from the value to
-				// an *slog.Source, it means the caller logged
-				// another attribute with a key of `caller`.
-				// Prevent duplicate keys (necessary for proper
-				// JSON) by renaming the key to
-				// `logged_caller`.
-				a.Key = reservedKeyPrefix + key
-			}
-		case slog.LevelKey:
-			if lvl, ok := a.Value.Any().(slog.Level); ok {
-				a.Value = slog.StringValue(strings.ToLower(lvl.String()))
-			} else {
-				// If we can't cast the any from the value to
-				// an slog.Level, it means the caller logged
-				// another attribute with a key of `level`.
-				// Prevent duplicate keys (necessary for proper
-				// JSON) by renaming the key to `logged_level`.
-				a.Key = reservedKeyPrefix + key
-			}
-		default:
-		}
-
-		return a
-	}
-	defaultReplaceAttrFunc = func(groups []string, a slog.Attr) slog.Attr {
-		key := a.Key
-		switch key {
-		case slog.TimeKey:
-			if t, ok := a.Value.Any().(time.Time); ok {
-				a.Value = slog.TimeValue(t.UTC())
-			} else {
-				// If we can't cast the any from the value to a
-				// time.Time, it means the caller logged
-				// another attribute with a key of `time`.
-				// Prevent duplicate keys (necessary for proper
-				// JSON) by renaming the key to `logged_time`.
-				a.Key = reservedKeyPrefix + key
-			}
-		case slog.SourceKey:
-			if src, ok := a.Value.Any().(*slog.Source); ok {
-				a.Value = slog.StringValue(filepath.Base(src.File) + ":" + strconv.Itoa(src.Line))
-			} else {
-				// If we can't cast the any from the value to
-				// an *slog.Source, it means the caller logged
-				// another attribute with a key of `source`.
-				// Prevent duplicate keys (necessary for proper
-				// JSON) by renaming the key to
-				// `logged_source`.
-				a.Key = reservedKeyPrefix + key
-			}
-		case slog.LevelKey:
-			if _, ok := a.Value.Any().(slog.Level); !ok {
-				// If we can't cast the any from the value to
-				// an slog.Level, it means the caller logged
-				// another attribute with a key of `level`.
-				// Prevent duplicate keys (necessary for proper
-				// JSON) by renaming the key to
-				// `logged_level`.
-				a.Key = reservedKeyPrefix + key
-			}
-		default:
-		}
-
-		return a
-	}
+	defaultWriter = os.Stderr
 )
 
-// AllowedLevel is a settable identifier for the minimum level a log entry
-// must be have.
-type AllowedLevel struct {
-	s   string
+// Level controls a logging level, with an info default.
+// It wraps slog.LevelVar with string-based level control.
+// Level is safe to be used concurrently.
+type Level struct {
```

Comment:
As mentioned in the description - this (rename) is a breaking change, let me know if this is ok.